src/hazmat/log2.rs

```diff
@@ -0,0 +1,106 @@
+// Copied from the `libm` crate, which is licensed under the MIT license.
```

Comment:
@fjarri Adding this code like this is not great but without a high-quality log2 implementation it all kind of falls apart and precision is terrible. Our options are limited:

- Do what I did here, vendor the code. It's small, unlikely to change and is not exported.
- Add a dependency on `libm` and import `log2` from there; likely we'd use the `libm` feature from `num-traits`. Note that while `num-traits` is a dependency of `crypto-bigint`, it does not enable the `libm` feature.
- Wait for https://github.com/rust-lang/rust/issues/137578 to land on stable (could be a long wait...)

Thoughts?

Reply:
Given the size of the code we need to vendor (including what's already there for the FIPS stuff), I think a dependency on `libm` is a better choice.

Reply:
Fair enough.

Reply:
> Given the size of the code we need to vendor (including what's already there for the FIPS stuff), 

Just a note on this: it's 83 lines of code, so I'd argue it's not a lot; also, the `libm` original methods are not `const`, so we're not getting rid of `floor` and `float_sqrt` from `hazmat::float` without un-constify the FIPS stuff.

Reply:
Good point. I wonder why it isn't const in `libm`.

Reply:
Guessing it's just an oversight. Or the code pre-dates the possibility of doing so (e.g. for `log2` git blame says 7 years...)

Reply:
Evidently they don't want to do it, at least not anytime soon: https://github.com/rust-lang/compiler-builtins/issues/893

Reply:
That was an interesting zulip thread. Ok, let's check again in 5-10 years! ;)